### PR TITLE
loongarch: add __clear_cache builtin implementation

### PR DESCRIPTION
**Pull request by yxd-ym**
- add __clear_cache builtin implementation

### CODE REVIEW

File: src/Compilation/Config.zig [hunk 1]
```diff
@@ -302,6 +302,12 @@ pub fn resolve(options: Options) ResolveError!Config {
             break :b false;
         }
 
+        if (target.cpu.arch.isLoongArch()) {
+            // FIXME:
+            // loongarch LTO might have problem on LLVM 18.
```
Comment (alexrp) [inline src/Compilation/Config.zig:307]: This needs more details to be mergeable: What exactly is the problem? Is it expected to be fixed in 19?

File: src/codegen/llvm.zig [hunk 2]
```diff
@@ -11786,6 +11786,7 @@ fn backendSupportsF16(target: std.Target) bool {
         .mipsel,
         .mips64,
         .mips64el,
+        .loongarch64, // FIXME: See: https://github.com/llvm/llvm-project/issues/93894
```
Comment (alexrp) [inline src/codegen/llvm.zig:11789]: I expect that we'll be upgrading to LLVM 19 in this release cycle, assuming there are no issues blocking the upgrade. So I'm not sure it's worth it to apply workarounds for issues we know will be fixed in 19.

**Comment by alexrp** [comment]
Changes in `clear_cache.zig` LGTM.

File: src/Compilation/Config.zig (hunk 1 above)
Comment (yxd-ym) [reply src/Compilation/Config.zig]: I checked the source again. The problem was in GCC not LLVM.

https://gcc.gnu.org/bugzilla/show_bug.cgi?id=113233

Removed this code.

File: src/codegen/llvm.zig (hunk 2 above)
Comment (yxd-ym) [reply src/codegen/llvm.zig:11789]: OK, remove it.

**Review by alexrp — APPROVED** [review]

**Comment by andrewrk** [comment]
Please run `zig build fmt` and it will fix the CI failures for you :-)

**Pull request merged**